### ISSUE TITLE
Specification-App Screen: redesign button alignment for Delete screen/Define screen/Save

### ISSUE THREAD

**Issue by fnisen** [appirio-tech/connect-app#767]
![screen shot 2017-03-09 at 10 50 18 am](https://cloud.githubusercontent.com/assets/12552247/23776393/ec2338a6-04e2-11e7-8216-ce09ea8e13d9.png)

Have ended up with a cluster of buttons that's kind of confusing.

**Comment by vic-topcoder** [comment]
Spec here: https://drive.google.com/drive/folders/0B1LbmKnex41fRDc3Y0ZxOGRCaHc?usp=sharing
![transition-v1-v2-screen-repeater](https://cloud.githubusercontent.com/assets/8484061/23777834/ea8713ce-04eb-11e7-9f9b-aa6779c01d53.png)

**Interactions**:

Let's display a small button "Delete" on the screen header;
When there is no screen defined (not all required elements are populated), the button is disabled;
When we have one screen definition, the button will be active, on press it'll reset the form values.
When we have 2+ screens, the button will remove the section without warning. The order of screens should be updated to reflect the change (Deleting screen 2 should make screen 3 to display 2, and so forth).

**Comment by vikasrohit** [comment]
>When we have one screen definition, the button will be active, on press it'll reset the form values.
I think it would create confusion and may set wrong expectation of users.  We should always(whether empty or filled) delete the screen from the defined screens list. And in any case we should never delete the screen from the database until user click Save Changes button.

**Comment by vic-topcoder** [comment]
@vikasrohit - yes, deleting form the DB should not be done until save. Regarding 1 screen confusion - I don't think it will create confusion if we have an animation/transition that removes the screen and leaves an empty element to fill. Due to the deadline and scope I'd suggest we give it a try in the simplest way possible. We want to enhance the screen definition process in the future to resemble more the feature picker, so we know we have to redesign the piece.

@fnisen: please chime in

**Comment by fnisen** [comment]
@vikasrohit @vic-topcoder 
Regarding Vic's point: to me there isn't much of a difference between the two behaviors described. The App Screens section is a requirement. The user will have to fill out at least 1 screen before submitting. So, there's really no reason for them to ever have no screens shown, even if they delete the last screen. After they delete the last screen, they would have to click the "Define screen" button again and restart. I say let's go for Vic's behavior, unless the technical complexity is much greater

Vikas, I'm also seeing a number of bugs with how "App Screens" is working and wanted to mention them here, so you can include as you make Vic's changes.
1. Bug with adding screens: if I first create a project, select a screen number, and then go to the App Screens section, the button to add screens is greyed out. Saving doesn't help. The only way I can add screens is by filling out the entire first section, which I wouldn't expect. Can you please change the behavior so that as soon as you select a number of screens, you can start adding new ones in the App Screens section? If needed, you can add the condition that the user "Save." 
2. Related to my first point, we can start the user off with 1 screen and just grey out the "Define another screen" option until they tell us how many screens they want designed. If it's easier to not start them off with any screens, let's call the button "Define screen" (because initially there are no screens)
3. The "Save" button in the App Screens section is turned off when that section is not passing validation. But, Save is still active in other sections. We should be consistent--either have all the buttons off or all of them on. (We should keep the "Submit for Review" button off until all sections pass validation).


Notes and Project Files not required

**Comment by fnisen** [comment]
Related issue: https://github.com/appirio-tech/connect-app/issues/751

**Comment by gondzo** [comment]
@fnisen #1 (Bug with creating screens) is already fixed (you can add new screens as soon as you save number of screens selection). I'm creating a new issue with your points #2 and 3 above. Let's leave this one just for button placement update.

**Comment by vikasrohit** [comment]
Thanks @gondzo

**Comment by fnisen** [comment]
@vikasrohit, @gondzo wanted to clarify something. When I make comments like this: 
> The "Save" button in the App Screens section is turned off when that section is not passing validation. But, Save is still active in other sections. We should be consistent--either have all the buttons off or all of them on. (We should keep the "Submit for Review" button off until all sections pass validation).

I'm expecting that you'll assess the current application functionality and make a recommendation as to how best to handle this. This is the reason I initially assigned this to Vikas and it means we can't simply put the issue into the bug bash. To me the bug bash is for straight-forward fixes--"this is what's wrong, this is how it should work." If you don't have time or want me to make the call on certain functionality, please let me know, and I'll do that. I'm trying to keep my comments a little open-ended so you can also be engaged in determining how things should work.

**Comment by vikasrohit** [comment]
@fnisen sorry, if I have mis interpreted your comment, however, I think it is bug where the Save button is behaving differently for App Screens section and I had anticipated the reason as well and when I looked into the code reason was the same. However, fixing it was not straight forward to me because of code structure for that section. I tried yesterday to clean that up but I ended up with more messed code so I reverted my changes and asked Samir to divide that task into two separate issues and assign it Colin (as he had already opted for a very closely related issue #751). I was hoping that it would be done by my today morning and I could validate things after.
